### ISSUE TITLE
Unverified Breakpoint - debugging and breakpoints not working

### ISSUE THREAD

**Issue by mitja13** [Microsoft/vscode-react-native#450]
Breakpoints cannot be set (app is successfully lunched), and bunch of errors are displayed. Tried a lot of fix recipes, created a vanilla project, reinstall modules ... but the errors persists.

## Actual Behavior
1. Breakpoints appear grey with an "Unverified Breakpoint" note

## Expected Behavior
1. Breakpoints work

## Software versions
```
vscode: 1.12.2
vscode-react-native: 0.3.2
node: 6.10.3
npm: 4.2.0
react-native-cli: 2.0.1
react-native: 0.44.0
mac-os: 10.12.4
```

## Outputs
- Output of the Debug Console (View -> Toggle Debug Console):
```
** BUILD SUCCEEDED **

Installing build/Build/Products/Debug-iphonesimulator/quizzing.app
Launching org.reactjs.native.example.quizzing
org.reactjs.native.example.quizzing: 17831
[vscode-react-native] Finished executing: react-native run-ios --simulator iPhone 5s --project-path ios

[Error] [Warning] Multiple plist candidates found. Application may not be in debug mode.

Stack: Error: Multiple plist candidates found. Application may not be in debug mode.
    at Function.getWarning (/Users/Mitja/.vscode/extensions/vsmobile.vscode-react-native-0.3.2/out/common/error/errorHelper.js:21:16)
    at nodeFileSystem.readDir.then (/Users/Mitja/.vscode/extensions/vsmobile.vscode-react-native-0.3.2/out/common/ios/simulatorPlist.js:37:68)
    at _fulfilled (/Users/Mitja/.vscode/extensions/vsmobile.vscode-react-native-0.3.2/node_modules/q/q.js:834:54)
    at self.promiseDispatch.done (/Users/Mitja/.vscode/extensions/vsmobile.vscode-react-native-0.3.2/node_modules/q/q.js:863:30)
    at Promise.promise.promiseDispatch (/Users/Mitja/.vscode/extensions/vsmobile.vscode-react-native-0.3.2/node_modules/q/q.js:796:13)
    at /Users/Mitja/.vscode/extensions/vsmobile.vscode-react-native-0.3.2/node_modules/q/q.js:604:44
    at runSingle (/Users/Mitja/.vscode/extensions/vsmobile.vscode-react-native-0.3.2/node_modules/q/q.js:137:13)
    at flush (/Users/Mitja/.vscode/extensions/vsmobile.vscode-react-native-0.3.2/node_modules/q/q.js:125:13)
    at _combinedTickCallback (internal/process/next_tick.js:67:7)
    at process._tickCallback (internal/process/next_tick.js:98:9)[vscode-react-native] Starting debugger app worker.
[vscode-react-native] Established a connection with the Proxy (Packager) to the React Native application
```
- Output of the Developer Tools console (Help -> Toggle Developer Tools -> Select Console tab):
```
[Extension Host] Error: The module '/Users/Mitja/.vscode/extensions/vsmobile.vscode-react-native-0.3.2/node_modules/dtrace-provider/build/Release/DTraceProviderBindings.node'
was compiled against a different Node.js version using
NODE_MODULE_VERSION 48. This version of Node.js requires
NODE_MODULE_VERSION 53. Please try re-compiling or re-installing
the module (for instance, using `npm rebuild` or`npm install`).
    at process.module.(anonymous function) [as dlopen] (ELECTRON_ASAR.js:173:20)
    at Object.Module._extensions..node (module.js:598:18)
    at Object.module.(anonymous function) [as .node] (ELECTRON_ASAR.js:173:20)
    at Module.load (module.js:488:32)
    at tryModuleLoad (module.js:447:12)
    at Function.Module._load (module.js:439:3)
    at Function.Y.o._load (/Users/Mitja/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/node/extensionHostProcess.js:4:366378)
    at Module.require (module.js:498:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (/Users/Mitja/.vscode/extensions/vsmobile.vscode-react-native-0.3.2/node_modules/dtrace-provider/dtrace-provider.js:17:23)
d.logExtensionHostMessage @ extensionHost.ts:285
(anonymous) @ extensionHost.ts:139
emitTwo @ events.js:106
emit @ events.js:191
process.nextTick @ internal/child_process.js:752
_combinedTickCallback @ internal/process/next_tick.js:67
_tickCallback @ internal/process/next_tick.js:98
extensionHost.ts:285 [Extension Host] Error: Cannot find module './build/default/DTraceProviderBindings'
    at Function.Module._resolveFilename (module.js:470:15)
    at Function.Module._load (module.js:418:25)
    at Function.Y.o._load (/Users/Mitja/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/node/extensionHostProcess.js:4:366378)
    at Module.require (module.js:498:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (/Users/Mitja/.vscode/extensions/vsmobile.vscode-react-native-0.3.2/node_modules/dtrace-provider/dtrace-provider.js:17:23)
    at Object.<anonymous> (/Users/Mitja/.vscode/extensions/vsmobile.vscode-react-native-0.3.2/node_modules/dtrace-provider/dtrace-provider.js:42:3)
    at Module._compile (module.js:571:32)
    at Object.Module._extensions..js (module.js:580:10)
    at Module.load (module.js:488:32)
d.logExtensionHostMessage @ extensionHost.ts:285
(anonymous) @ extensionHost.ts:139
emitTwo @ events.js:106
emit @ events.js:191
process.nextTick @ internal/child_process.js:752
_combinedTickCallback @ internal/process/next_tick.js:67
_tickCallback @ internal/process/next_tick.js:98
extensionHost.ts:285 [Extension Host] Error: Cannot find module './build/Debug/DTraceProviderBindings'
    at Function.Module._resolveFilename (module.js:470:15)
    at Function.Module._load (module.js:418:25)
    at Function.Y.o._load (/Users/Mitja/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/node/extensionHostProcess.js:4:366378)
    at Module.require (module.js:498:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (/Users/Mitja/.vscode/extensions/vsmobile.vscode-react-native-0.3.2/node_modules/dtrace-provider/dtrace-provider.js:17:23)
    at Object.<anonymous> (/Users/Mitja/.vscode/extensions/vsmobile.vscode-react-native-0.3.2/node_modules/dtrace-provider/dtrace-provider.js:42:3)
    at Module._compile (module.js:571:32)
    at Object.Module._extensions..js (module.js:580:10)
    at Module.load (module.js:488:32)
d.logExtensionHostMessage @ extensionHost.ts:285
(anonymous) @ extensionHost.ts:139
emitTwo @ events.js:106
emit @ events.js:191
process.nextTick @ internal/child_process.js:752
_combinedTickCallback @ internal/process/next_tick.js:67
_tickCallback @ internal/process/next_tick.js:98
extensionHost.ts:285 [Extension Host] (node:18252) DeprecationWarning: os.tmpDir() is deprecated. Use os.tmpdir() instead.
```

**Comment by chenzhiyuan0402** [comment]
same issue with me.

**Comment by vladimir-kotikov** [comment]
@mitja13, thanks for reporting.

While the debug console output looks scary in fact this is just a warning message that states that extension wasn't able to determine correct plist file location to use to enable JS debug mode on simulator and hence it's likely the thing that causes issues w/ inactive breakpoints.

Could you please look into RN debug meny on simulator (after starting debugging from VSCode) and check whether JS debug is enabled and if it isn't - enable it. This will trigger app to reload code and most likely will enable breakpoints.

**Comment by carlthuringer** [comment]
I am also having this issue. I tried your advice, @vladimir-kotikov but it does not enable breakpoints. Strangely it seemed to be working before, but many changes later it is now broken.

**Comment by mitja13** [comment]
@vladimir-kotikov the debug is working now, but not sure what was changed. did reconstruct the project structure from a new init (or package update did it).

**Comment by mostafaeweda** [comment]
same here

**Comment by vladimir-kotikov** [comment]
@carlthuringer, @mostafaeweda, any chance that you would be able to provide a minimal app that reproduces that issue? I suspect it might happen due to specific structure of your iOS project, but I was not able to repro this.

**Comment by vladimir-kotikov** [comment]
@carlthuringer, @mostafaeweda, closing this. If you're still experiencing the problem feel free to reopen

**Comment by EduardJS** [comment]
Hello. Unfortunately, I'm facing the very same issue.

Adding a breakpoint, appears greyed out, with an Unverified Breakpoint note. 
Note that the debug connsole has successfully established a connection to the proxy packager.

Also, I don't have any errors inside Help > DevTools > Console, iOS app's running ok, removed ios/build files so the packager can create a fresh batch of files ..

Please advise

**Comment by vladimir-kotikov** [comment]
@EduardJS hello and thanks for letting us know. We haven't heard about any issues with debugging for some time so this might be something new.

Would you mind creating a separate issue and filing all fields in issue's template to help us diagnose your problem?

**Comment by deadrats** [comment]
Hi there, I had an "unverified" BP issue, but I realized that the file was actually a soft link (ln).
Setting BP in the real file was fine..

**Comment by svekl** [comment]
Hi @deadrats, thank you for your comment, can you, please, provide us some more info? What file did you have as a soft link? I'm asking because I was not able to make RN packager to work with soft links.

**Comment by deadrats** [comment]
Le Sat, 21 Oct 2017 12:14:31 +0000 (UTC),
Serge Svekolnikov <notifications@github.com> a écrit :

>Hi @deadrats, thank you for your comment, can you, please, provide us some more info? What file
>did you have as a soft link? I'm asking because I was not able to make RN packager to work with
>soft links. 
>

Yes, sure.
I forgot to mention it was on Linux. 

Files are:

john@TaGaDaJoHn:~/src/irc4$ ll *py
-rw-rw-r-- 1 john john     43 oct.  12 12:55 00.py
-rw-rw-r-- 1 john john  26047 oct.  17 18:17 app.py
-rwxrwxr-x 1 john john  30075 oct.  17 17:10 bot.py*
-rw-rw-r-- 1 john john  47208 oct.  17 17:32 cmd.py
lrwxrwxrwx 1 john john      6 janv. 14  2016 cmds.py -> cmd.py
-rw-rw-r-- 1 john john  22909 oct.   7 10:18 dfun.py
-rw-rw-r-- 1 john john   1708 sept. 21 16:04 dgen2.py
-rw-rw-r-- 1 john john  22505 oct.  11 12:31 dial.py
-rw-rw-r-- 1 john john  20591 oct.   1 17:09 div.py
-rw-rw-r-- 1 john john  19571 sept. 21 18:38 dobj.py
-rw-rw-r-- 1 john john   7102 juin  11 08:30 dold.py
-rw-rw-r-- 1 john john   1151 mars   2  2017 eliza.py
-rwxrwxr-x 1 john john    294 juil. 25 12:00 fltacc2.py*
-rwxrwxr-x 1 john john    243 mars  19  2017 fltacc.py*
-rw-rw-r-- 1 john john  29498 sept. 24 07:14 fun.py
-rw-rw-r-- 1 john john 157552 oct.  18 02:00 game.py
-rw-rw-r-- 1 john john  27260 oct.   5 18:39 gen.py
-rw-rw-r-- 1 john john  33187 sept. 13 16:15 hi.py
-rw-r--r-- 1 john john  13185 mai   10 17:37 ircbot.py
-rw-r--r-- 1 john john  48707 août   7 18:07 irclib.py
-rw-rw-r-- 1 john john  24990 sept. 22 20:23 irc.py
-rw-rw-r-- 1 john john  13871 août  28 21:27 lang.py
-rw-rw-r-- 1 john john  28278 mars  22  2017 metaphone.py
-rw-rw-r-- 1 john john  25327 oct.  20 13:24 net.py
-rw-rw-r-- 1 john john  28125 oct.  17 18:56 obj.py
lrwxrwxrwx 1 john john      2 juil. 11  2016 p2.py -> p2*
-rw-rw-r-- 1 john john   1220 août  18 21:06 params.py
-rw-rw-r-- 1 john john    659 mars  25  2016 parms.py
-rwxrwxr-x 1 john john  13692 oct.  17 18:43 psy.py*
-rw-rw-r-- 1 john john  27220 oct.   3 14:58 run.py
-rw-rw-r-- 1 john john  14603 oct.   8 08:44 scr.py
-rw-rw-r-- 1 john john 287015 oct.  20 14:05 toy.py
-rw-rw-r-- 1 john john  30224 oct.   8 19:57 trig.py
-rw-rw-r-- 1 john john  13338 oct.  17 19:32 utl.py
-rw-rw-r-- 1 john john  10955 août  10 11:56 vrac.1.py
-rw-rw-r-- 1 john john  17301 août   6 13:19 vrac.2.py
-rw-rw-r-- 1 john john  23620 sept. 28 10:16 vrac.3.py
-rw-rw-r-- 1 john john  23620 sept. 28 08:28 vrac.py

soft links are 'file1 -> file2' if you're not familiar w/ Linux;
I wasn't able to use "cmd.py" because it conflicts within "pdb", the standard low-level debugger.

So I made the soft link 'cmd.py -> cmds.py'
But seting the BP with code-insider did not work.. Until I set it at the right place !
(the actual file)


I realize this is doubly obolete ><
Thanks for all, have it easy

--
JoHn.
